src/workerd/jsg/resource.h

```diff
@@ -1492,12 +1549,14 @@ class ResourceWrapper {
     }
   }
 
-  template <bool isContext = false>
+  template <bool isContext = false, bool memoize = true>
   v8::Local<v8::FunctionTemplate> getTemplate(v8::Isolate* isolate, T*) {
     v8::Global<v8::FunctionTemplate>& slot = isContext ? contextConstructor : memoizedConstructor;
     if (slot.IsEmpty()) {
-      auto result = makeConstructor<isContext>(isolate);
-      slot.Reset(isolate, result);
+      auto result = makeConstructor<isContext, memoize>(isolate);
+      if (memoize) {
```

Comment:
Can this be `if constexpr (memoize) {`?

Reply:
yep, no difference as compiler will optimize it away but I generally like the if constexpr feature so lets use it 😄